### PR TITLE
Add support for 2.13.1

### PR DESCRIPTION
**Pull request by gabro**
Getting things ready.

As soon as 2.13.1 is out, we need to merge https://github.com/scalameta/scalameta/pull/1901, backpublish Scalameta 4.2.3 for 2.13.1 and then we can test this one.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -205,7 +205,17 @@ lazy val mtags = project
...
       "scalaCompilerVersion" -> scalaVersion.value
-    )
+    ),
+    unmanagedSourceDirectories in Compile ++= {
+      val sourceDir = (sourceDirectory in Compile).value
+      scalaVersion.value.split('.') match {
+        case Array(epic, major, patch) =>
+          List(
+            sourceDir / s"scala-$epic.$major",
+            sourceDir / s"scala-$epic.$major.$patch"
```
Comment (gabro) [inline build.sbt]: e.g. for Scala 2.12.8 add `scala-2.12` and `scala-2.12.8` dirs

File: build.sbt [hunk 2]
```diff
@@ -205,7 +205,17 @@ lazy val mtags = project
     buildInfoPackage := "scala.meta.internal.mtags",
     buildInfoKeys := Seq[BuildInfoKey](
       "scalaCompilerVersion" -> scalaVersion.value
-    )
+    ),
+    unmanagedSourceDirectories in Compile ++= {
```
Comment (olafurpg) [inline build.sbt]: Is this change needed anymore?

File: tests/cross/src/test/scala/tests/pc/MacroCompletionSuite.scala [hunk 3]
```diff
@@ -136,7 +136,7 @@ object MacroCompletionSuite extends BaseCompletionSuite {
     compat = Map(
       // NOTE(olafur): the presentation compiler returns empty results here in 2.12.9 and 2.13.0
       "2.12.9" -> "",
-      "2.13" -> ""
+      "2.13.0" -> ""
```
Comment (olafurpg) [inline tests/cross/src/test/scala/tests/pc/MacroCompletionSuite.scala]: Do I understand correctly that the 2.13 regressions for macros have been fixed in 2.13.1?

Comment (gabro) [reply tests/cross/src/test/scala/tests/pc/MacroCompletionSuite.scala]: yes, and also 2.12.10 (we hadn't noticed it at the time)

File: build.sbt (hunk 2 above)
Comment (gabro) [reply build.sbt]: I think so, `AdaptedReporter` is present in 2.13.0 but it's gone in 2.13.1. I welcome alternative ideas :D

File: mtags/src/main/scala/scala/meta/internal/pc/Keywords.scala [hunk 4]
```diff
@@ -115,7 +116,7 @@ trait Keywords { this: MetalsGlobal =>
   private def isBlock(enclosing: List[Tree]): Boolean =
     enclosing match {
       case Ident(_) :: Block(_, _) :: _ => true
-      case _ => false
+      case other => false
```
Comment (gabro) [inline mtags/src/main/scala/scala/meta/internal/pc/Keywords.scala]: why binding `other`?

Comment (tgodzik) [reply mtags/src/main/scala/scala/meta/internal/pc/Keywords.scala]: Fixed, sorry!
